### DISCUSSION TITLE
Cleaning Full Route Cache on deployments

### DISCUSSION THREAD

**Discussion by jnm733** [caching-tools/next-shared-cache#334] [Q&A]
Like the official doc indicate, I want to clean my Full Route Cache on every deployment.

https://nextjs.org/docs/app/building-your-application/caching#invalidation

These records have the field 'kind: "PAGE"' on Redis values. Which is the easily way to clean these records automatically on our deploy pipeline maintaining data cache records?

Thanks!

**Comment by better-salmon** [comment]
Hello @jnm733! Sorry for the delay. When deploying the new version, do you want to clean all cache entries except `kind: 'FETCH'`?

**Reply by jnm733** [reply]
Hi!

Yes, it would be perfect to clean Full Route Cache between deployments without affect to the FETCH caches that stay the same.

**Reply by better-salmon** [reply]
For now, there is no built-in solution. I can offer a concept of a solution. You can create a JS file that connects to your remote storage without CacheHandler, finds every non-`FETCH` value, and deletes the corresponding entries. Then, you can add this JS file to `postbuild` script in `package.json`: 

```json
{
  "scripts": {
    "build": "next build",
    "postbuild": "node clean-full-route-cache.js"
  }
}
```

This is just a concept. I need more time to come up with a proper solution.
